url8af84d296f_mrcosta_gantt_generator_tgz-pJ8-org_gantt_generator_mrcpsp_RunJ8/src/org/swiftgantt/common/SwingComImageGenerator.java

```diff
@@ -0,0 +1,190 @@
+			}
+		}catch (Exception ex) {
+			logger.error("generateJPEGImage(Dimension, JComponent[], Rectangle[], String)", ex); //$NON-NLS-1$
+			throw new IOException(ex.getMessage() + ex.getStackTrace());
+		}
+	}
+	
+	
+	/**
+	 * 
+	 * @param component
+	 * @param filePath
+	 * @throws Exception
+	 */
+	public void genPNGImage(JComponent component, String filePath) throws IOException{
+		Rectangle bounds = component.getBounds();
+		this.genPNGImage(new Dimension(bounds.width, bounds.height), new JComponent[]{component}, new Rectangle[]{bounds}, filePath);
+	}
+
+	/**
+	 * 
+	 * @param size
+	 * @param components
+	 * @param areas
+	 * @param filePath
+	 * @throws IOException
+	 */
+	public void genPNGImage(Dimension size, JComponent[] components, Rectangle[] areas, String filePath) throws IOException{
+		logger.debug("Generate JPEG image file");
+		if(components.length != areas.length){
+			throw new RuntimeException("The specified components is not fulfil the requirement");
+		}
+		if(filePath == null){
+			String dir = System.getProperty("user.dir") + System.getProperty("file.separator") + "image";
+			filePath = dir + System.getProperty("file.separator") + "ganttchart.jpg";
+		}
+		checkAndCreateDirForFilePath(filePath);
+		BufferedImage image = new BufferedImage(size.width, size.height, BufferedImage.TYPE_INT_RGB);
+		Graphics imageGraphic = image.getGraphics();
+		// Background is white.
+		imageGraphic.setColor(Color.WHITE);
+		imageGraphic.fillRect(0, 0, size.width, size.height);
+
+		for (int i = 0; i < components.length; i++) {
+			JComponent curCom = components[i];
+			Rectangle bounds = areas[i];
+			if (logger.isDebugEnabled()) {
+				logger.debug("Start to paint component " + curCom.getClass().getName() + " to image:" + bounds); //$NON-NLS-1$ //$NON-NLS-2$
+			}
+			curCom.paint(imageGraphic);
+			if(bounds.x > 0 || bounds.y > 0){
+				// Copy may be not effective.
+				imageGraphic.copyArea(0, 0, bounds.width, bounds.height, bounds.x, bounds.y);
+			}
+		}
+		try{
+			File imgFile = new File(filePath);
+			ImageOutputStream ios = new FileImageOutputStream(imgFile);
+			ImageWriterSpi iwspi = new PNGImageWriterSpi();
+			PNGImageWriter pngImageWriter = new PNGImageWriter(iwspi);
+			pngImageWriter.setOutput(ios);
+			pngImageWriter.write(image);
+			if (logger.isDebugEnabled()) {
+				logger.debug("Image:" + filePath + " generated"); //$NON-NLS-1$ //$NON-NLS-2$
+			}
+			ios.close();
+		}catch (Exception ex) {
+			logger.error("generatePNGImage(Dimension, JComponent[], Rectangle[], String)", ex); //$NON-NLS-1$
+			throw new IOException(ex.getMessage() + ex.getStackTrace());
+		}
+	}
+	
+	/**
+	 * 
+	 * @param filePath
+	 */
+	public static void checkAndCreateDirForFilePath(String filePath){
+		File file = new File(filePath);
+		File dir = file.getParentFile();
+		if (!dir.exists()) {
+			dir.mkdirs();
```

Comment:
Recommendation generated by Amazon CodeGuru Reviewer. Leave feedback on this recommendation by replying to the comment or by reacting to the comment using emoji.

Confirm that `mkdirs` was successful. Handle any failures that happened. If you fail silently, then you might encounter errors that are harder to debug. See the following diffs on GitHub that make this code change: [diff1](https://github.com/qunarcorp/qmq/commit/12f1d9d6a57a02bc2415cb2af9f4388a77b12c02#diff-6b67be6e17f24fdeddb7c725ac8ec5600d6e63be0bdca0963a83d530e6d55e35L36), [diff2](https://github.com/apache/jmeter/commit/1ec61206779b3f84df18516682b4791bf015178f#diff-cd74304763c097fbc085ac2bc952b46ab3843ac825cacacad89e8064296e955aL330)

Learn more: [File.mkdirs in the Java Platform, Standard Edition 8 API Specification](https://docs.oracle.com/javase/8/docs/api/java/io/File.html#mkdirs--)